imports/plugins/core/accounts/server/methods/addressBookUpdate.js

```diff
@@ -86,22 +90,14 @@ export default function addressBookUpdate(address, accountUserId, type) {
     }
   });
 
-  // Run afterAccountsUpdate hook to update Accounts Search
-  Hooks.Events.run("afterAccountsUpdate", userId, {
-    accountId: account._id,
+  const updatedAccount = Accounts.findOne({ userId });
+  Promise.await(appEvents.emit("afterAccountUpdate", {
+    updatedAccount,
+    updatedBy: authUserId,
     updatedFields
-  });
+  }));
 
   // If the address update was successful, then return the full updated address
-  if (updatedAccountResult === 1) {
-    // Find the account
-    const updatedAccount = Accounts.findOne({
-      userId
-    });
-
-    // Pull the updated address and return it
-    return updatedAccount.profile.addressBook.find((updatedAddress) => address._id === updatedAddress._id);
-  }
-
-  throw new ReactionError("server-error", "Unable to update account address");
+  // Pull the updated address and return it
+  return updatedAccount.profile.addressBook.find((updatedAddress) => address._id === updatedAddress._id);
```

Comment:
Do we need to check that `updatedAccount.profile.addressBook` exists here?

Reply:
Same answer